File: prac3-6/pila.py
class Pila:
    def __init__(self):
        self.items = []

    def push(self, dato):
        self.items.append(dato)

    def pop(self):
        if self.items:
            return self.items.pop()
        return "Pila vacía"

    def top(self):
        if self.items:
            return self.items[-1]
        return "Pila vacía"

    def mostrar(self):
        print(self.items[::-1])

pila = Pila()

for x in [5,10,15,20,25]:
    pila.push(x)

print("Pila inicial:")
pila.mostrar()

print("Elemento eliminado:", pila.pop())
print("Elemento eliminado:", pila.pop())

print("Tope actual:", pila.top())

print("Pila final:")
pila.mostrar()

File: prac3-6/pila.cpp
#include <iostream>
#include <stack>
using namespace std;

int main() {
    stack<int> pila;

    pila.push(5);
    pila.push(10);
    pila.push(15);
    pila.push(20);
    pila.push(25);

    pila.pop();
    pila.pop();

    cout << "Tope actual: " << pila.top() << endl;

    return 0;
}